### DISCUSSION TITLE
Any way to reuse TCP connections to redis similar to the keep-alive configuration option for http remote storage?

### DISCUSSION THREAD

**Discussion by carterols** [ccache/ccache#1520] [Questions & answers]
We have a twemproxy remote storage that talks with multiple redis instances. During a build, I am seeing many TCP connections established to the same twemproxy remote, but are then destroyed and reestablished many times. I see that the http remote storage has a "keep-alive" option for tcp keep-alive, and I was wondering if this could also be applied to the redis remote storage?

**Comment by jrosdahl** [comment]
Unfortunately there is no way to improve the Redis backend given the current architecture. Each ccache invocation needs to open a new connection to the remote server, which happens both with the Redis and the HTTP backend. The HTTP keep-alive option is about reusing the same connection for multiple queries performed by a single ccache call, but this is already done by the Redis implementation.

A way to improve this is described in #1214.

**Reply by carterols** [reply]
This is a really cool idea. I hope this can become the future of remote storage for ccache!

**Comment by afbjorklund** [comment]
Did you try using unix domain sockets for the local proxy? With the `redis+unix:` syntax. It should have a lower setup time.

```diff
 pool:
-  listen: 127.0.0.1:6379
+  listen: /var/run/nutcracker.sock
   redis: true
   ...
```

Nutcracker/twemproxy is supposed to keep the connection to the remote cache(s) open for you, between the ccache calls...

Also https://github.com/ccache/ccache/wiki/Redis-storage
